content/api/versions/v0.8/rest-sdk/encryption.textile

```diff
@@ -0,0 +1,222 @@
+---
+title: Encryption
+meta_description: "Client Library SDK REST API Reference Crypto documentation."
+meta_keywords: "Ably, Ably REST, API Reference, REST SDK, REST interface, REST API, Encryption, Crypto"
+section: api
+index: 90
+languages:
+  - javascript
+  - nodejs
+  - php
+  - python
+  - ruby
+  - java
+  - swift
+  - objc
+  - csharp
+jump_to:
+  API Reference:
+    - getDefaultParams#get-default-params
+    - generateRandomKey#generate-random-key
+  Related types:
+    - Channel Options
+    - CipherParams#cipher-params
+---
+
+inline-toc.
+  Methods:
+    - getDefaultParams#get-default-params
+    - generateRandomKey#generate-random-key
+  Related types:
+    - Channel Options
+    - CipherParams#cipher-params
+
+The <span lang="default">@Ably.Rest.Crypto@</span><span lang="ruby">@Ably::Util::Crypto@</span><span lang="php">@Ably\Utils\Crypto@</span><span lang="python">@ably.util.crypto@</span><span lang="java">@io.ably.lib.util.crypto@</span><span lang="swift,objc">@ARTCrypto@</span><span lang="csharp"> @IO.Ably.Encryption.Crypto@</span> object exposes the following public methods:
+
+h2(#methods). Methods
+
+h6(#get-default-params).
+  default:  getDefaultParams
+  ruby:     get_default_params
+  python:   get_default_params
+  csharp:   GetDefaultParams
+
+bq(definition).
+  jsall:   "CipherParams":#cipher-params Crypto.getDefaultParams(Object params)
+  ruby:    "CipherParams":#cipher-params Crypto.get_default_params(Hash params)
+  python:  "CipherParams":#cipher-params Crypto.get_default_params(Dict params)
+  php:     "CipherParams":#cipher-params Crypto.getDefaultParam(Array params)
+  java:    "CipherParams":#cipher-params Crypto.getDefaultParams(Param[] params)
+  csharp:  "CipherParams":#cipher-params GetDefaultParams(byte[] key = null, byte[] iv = null, CipherMode? mode = null)
+  swift,objc: getDefaultParams(values: [NSObject : AnyObject]) -> "ARTCipherParams":#cipher-params
+
+<span lang="default">This call obtains a "@CipherParams@":#cipher-params object using the values passed in (which must be a subset of @CipherParams@ fields that at a minimum includes a @key@), filling in any unspecified fields with default values, and checks that the result is a valid and self-consistent.</span><span lang="csharp">This call takes a key, an initialization vector (iv) and a Cipher mode. There is also on override which accepts the @key@ and @iv@ as base64 encoded strings. It will validate the passed values and generate @CipherParams@</span>
+
+You will rarely need to call this yourself, since the client library will handle it for you if you specify @cipher@ params when initializing a channel (as in the example "at the top":#getting-started)<span lang="jsall"> or when setting channel options with @channel#setOptions@</span>.
```

Comment:
Same comment on #getting-started as above - I guess it might be in all versions?